### DISCUSSION TITLE
Build issues in x86_64-darwin

### DISCUSSION THREAD

**Discussion by wineee** [cpeditor/cpeditor#1026] [Q&A] [answered]
I try to packaged cpeditor 6.9.3 in Nix，a  package manager for Linux and other Unix systems. 
Build successful on Linux， but failed to build on x86_64-darwin, 
> [116/116] Linking CXX executable cpeditor.app/Contents/MacOS/cpeditorYJK5W5UP7/qrc_macos-resources.cpp.o
installing
install flags: install
ninja: error: unknown target 'install'
builder for '/nix/store/p8njjv1jj63s1r4248svvkq233l43n8x-cpeditor-6.9.3.drv' failed with exit code 1
error: build of '/nix/store/p8njjv1jj63s1r4248svvkq233l43n8x-cpeditor-6.9.3.drv' failed

build log：https://logs.nix.ci/?key=nixos/nixpkgs.148304&attempt_id=10e88c7a-3251-493e-a85b-75ad00068abd

I'm not sure what's wrong, seem cpeditor supports MacOS

**Comment by coder3101** [comment, marked as answer]
Install target is generated for Non Darwin UNIX Operating System only. 

See: https://github.com/cpeditor/cpeditor/blob/2eeed49e90e16dfdd37ce55676ce28dde5a9fc0d/CMakeLists.txt#L291

It is because, cpeditor is GUI application built as macOS bundle in apple whereas in linux its a typical ELF binary. 

Are you sure that this is how macOS GUI apps are packaged in NIX? Also you might want to take a look at https://github.com/NixOS/nix/issues/956

**Reply by coder3101** [reply]
@swift-zym and @ouuan Do you think we should add an install target for macOS which just installs(copies) cpeditor bundle to / Application or a given prefix?

**Comment by wineee** [comment]
Actually, I don't use MacOS, I think that set platforms to linux only in nix is better，unless some darwin user want to maintain it.

**Reply by coder3101** [reply]
> Actually, I don't use MacOS, I think that set platforms to linux only in nix is better，unless some darwin user want to maintain it.

Exactly, seems like nix needs some manual intervention for correctly installing GUI apps on macOS. If you don't use it, don't package for that platform.
